examples/llm/src/models/layers/attention.py

```diff
@@ -179,16 +196,31 @@ def triton_flash_attn_fn(
         raise NotImplementedError(
             f'attn_impl: triton cannot return attn weights.')
 
+    if query_padding_mask is not None:
+        query = query.masked_fill(~query_padding_mask.unsqueeze(-1), 0)
     if key_padding_mask is not None:
-        b_size, s_k = key_padding_mask.shape
+        key = key.masked_fill(~key_padding_mask.unsqueeze(-1), 0)
+        value = value.masked_fill(~key_padding_mask.unsqueeze(-1), 0)
+
+    if query_padding_mask is not None or key_padding_mask is not None:
+        b_size, s_q, s_k = query.size(0), 1, 1
+        if query_padding_mask is not None:
+            s_q = query_padding_mask.size(1)
+        if key_padding_mask is not None:
+            s_k = key_padding_mask.size(1)
 
         if attn_bias is not None:
             attn_bias = attn_bias.expand(b_size, -1, -1, -1)
         else:
-            attn_bias = query.new_zeros(b_size, 1, 1, s_k)
+            attn_bias = query.new_zeros(b_size, 1, s_q, s_k)
```

Comment:
aren't s_q and s_k undefined if query_padding_mask and key_padding_mask are not None

Reply:
[They default to 1](https://github.com/mosaicml/examples/pull/244/files/0d459af1ac60055092e1e0c972a9e29d5b26fe55#diff-22824d61d86c454e9e92d5ea2b1a233d8d972160e01d2261f3dd8f18c307c4f5R206)
and everything works

Reply:
They won't default to one when both query_padding_mask and key_padding_mask are given and will be undefined. Do we have a test for this case?

Reply:
are s_k or s_q used outside of the `if query_padding_mask is not None or key_padding_mask is not None:` context?

Reply:
nvm. I misread line 215 as outside the if context.